interface/ceed-operator.c

```diff
@@ -304,8 +304,58 @@ int CeedOperatorAssembleLinearQFunction(CeedOperator op, CeedVector *assembled,
       return CeedError(ceed, 1, "At least one non-collocated basis required");
     // LCOV_EXCL_STOP
   }
-  ierr = op->AssembleLinearQFunction(op, assembled, rstr, request);
-  CeedChk(ierr);
+  if (op->AssembleLinearQFunction) {
+    ierr = op->AssembleLinearQFunction(op, assembled, rstr, request);
+    CeedChk(ierr);
+  } else { // Delegate to /cpu/self/ref/serial
+    Ceed ceedref;
+    ierr = CeedInit("/cpu/self/ref/serial", &ceedref); CeedChk(ierr);
```

Comment:
It isn't the first use of this pattern (thus perhaps out of scope for this PR), but should backends be responsible for identifying their delegate (`ceedref`), instead of losing that symmetry here in interface code? It could be stored in `ceed->ceedref`.

Reply:
Backends do generally specify their own delegate and are responsible for their creation (but not destruction). A `ceed` can specify a default delegate as well as delegates for specific objects; however, delegates are treated as all-or-nothing with respect to specific objects. The code is often interdependent enough that I have been assuming that we can't mix and match functions from different `ceed`s with the same object.

```
struct Ceed_private {
  const char *resource;
  Ceed delegate;
  Ceed parent;
  objdelegate *objdelegates;
  int objdelegatecount;
...
```

We don't currently have a way for any backend to delegate only `CeedOperatorAssembleLinearQFunction` while implementing all other `CeedOperator` functions.

This is the first use of a single function delegation. Whatever we adopt here I'd like to also use for `AssembleLinearDiagonal` and `CreateElementFDMInverse`, so it seems like the right time to change it if we want to.

I can add
1) carrying around this operator+qfunction clone for the life of the operator and/or
2) letting the backend specify this 'delegate' (maybe 'fallback'? we're overusing the phrase 'delegate' to mean two/three different things here) but default to `/cpu/self/ref/serial`

if that would be more preferable/flexible/extensible/ect

Reply:
@jedbrown, I refactored to make it clear that this is different that our typical delegation mechanism and to allow backends to select something other than `/cpu/self/ref/serial` if desired. What do you think?